programs/clearing_house/src/controller/orders.rs

```diff
@@ -1254,11 +1256,23 @@ pub fn fulfill_order_with_amm(
     let user_position_delta =
         get_position_delta_for_fill(base_asset_amount, quote_asset_amount, order_direction)?;
 
-    update_amm_and_lp_market_position(
-        market,
-        &user_position_delta,
-        cast_to_i128(fee_to_market_for_lp)?,
-    )?;
+    if split_with_lps {
+        update_amm_and_lp_market_position(market, &user_position_delta, fee_to_market_for_lp)?;
```

Comment:
dont like how depending on the function you pass in the user_position_delta OR the amm_position_delta. the corret logic should be handled inside